src/Builder.js

```diff
@@ -14,10 +14,43 @@ var pexprs = require('./pexprs');
 function Builder() {}
 
 Builder.prototype = {
+  currentDecl: null,
+
   newGrammar: function(name) {
     return new GrammarDecl(name);
   },
 
+  grammar: function(metaInfo, name, superGrammar, defaultStartRule, rules) {
+    var gDecl = new GrammarDecl(name);
+    if (superGrammar) {
+      gDecl.withSuperGrammar(this.fromRecipe(superGrammar));
+    }
+    if (defaultStartRule) {
+      gDecl.withDefaultStartRule(defaultStartRule);
+    }
+    if (metaInfo && metaInfo.source) {
+      gDecl.withSource(metaInfo.source);
+    }
+
+    var self = this;
+    this.currentDecl = gDecl;
+    Object.keys(rules).forEach(function(ruleName) {
+      var ruleRecipe = rules[ruleName];
+
+      var action = ruleRecipe[0]; // define/extend/override
+      // var metaInfo = ruleRecipe[1]; // TODO: us and handle meta-info
```

Comment:
One thing I need for the visualizer, but hadn't implemented yet, is the "definition interval" of a rule. E.g., given a grammar `G { foo = digit }`, the definitionInterval would be `[4, 15]`. So it would definitely be useful to be able to have rule metadata.